toolkit/scripts/toolchain/create_toolchain_in_container.sh

```diff
@@ -8,28 +8,38 @@ set -e
 MARINER_BUILD_DIR=$1
 MARINER_SPECS_DIR=$2
 MARINER_SOURCE_URL=$3
+INCREMENTAL_TOOLCHAIN=$4
+ARCHIVE_TOOL=$5
 
-./toolchain_verify.sh $MARINER_BUILD_DIR
-
-# Cleanup
-docker images -a
-docker ps -a
-# docker system prune -f
-# docker rmi $(docker images -a -q)
-# docker rmi $(docker history marinertoolchain -q)
-
-# CPIO patch
-cp -v $MARINER_SPECS_DIR/cpio/cpio_extern_nocommon.patch ./container
-cp -v $MARINER_SPECS_DIR/cpio/CVE-2021-38185.patch ./container
-# Coreutils aarch64 patch
-cp -v $MARINER_SPECS_DIR/coreutils/coreutils-fix-get-sys_getdents-aarch64.patch ./container
-# Binutils readonly patch
-cp -v $MARINER_SPECS_DIR/binutils/linker-script-readonly-keyword-support.patch ./container/linker-script-readonly-keyword-support.patch
-# RPM LD_FLAGS patch
-cp -v $MARINER_SPECS_DIR/rpm/define-RPM_LD_FLAGS.patch ./container/rpm-define-RPM-LD-FLAGS.patch
-
-# Create .bashrc file for lfs user in the container
-cat > ./container/.bashrc << EOF
+# Grab an identity for the raw toolchain components so we can avoid rebuilding it if it hasn't changed
+sha_component_tag=$(sha256sum ./container/toolchain-sha256sums ./container/toolchain_build_in_chroot.sh ./create_toolchain_in_container.sh | sha256sum | cut -d' ' -f1 )
+export sha_component_tag
+
+# Check if we already have a committed container with the same sha_component_tag
+if [ "$INCREMENTAL_TOOLCHAIN" == "y" ] && [ -z "$(docker images -q marinertoolchain_populated:${sha_component_tag} 2>/dev/null)" ]; then
```

Comment:
>if [ "$INCREMENTAL_TOOLCHAIN" == "y" ] [](http://example.com/codeflow?start=0&length=38)

Wouldn't this break for clean repos, where we build the toolchain but without `INCREMENTAL_TOOLCHAIN=y`?

Reply:
Hmm, you are right I think the logic is inverted. It intent is to never use the cached version unless `INCREMENTAL_TOOLCHAIN=y` is set.

I think the correct version should be:

```bash
if [ "$INCREMENTAL_TOOLCHAIN" != "y" ] || [ -z "$(docker images -q marinertoolchain_populated:${sha_component_tag} 2>/dev/null)" ]; then
...
```

Reply:
Hmm, you are right I think the logic is inverted. It intent is to never use the cached version unless `INCREMENTAL_TOOLCHAIN=y` is set.

I think the correct version should be:

```bash
if [ "$INCREMENTAL_TOOLCHAIN" != "y" ] || [ -z "$(docker images -q marinertoolchain_populated:${sha_component_tag} 2>/dev/null)" ]; then
...
```